### ISSUE TITLE
qfile-unpacker error during copying of lot of files

### ISSUE THREAD

**Issue by 100111001** [QubesOS/qubes-issues#5370]
**Qubes OS version**
R4.0

**Affected component(s) or functionality**
qvm-copy-to-vm

**Brief summary**
During copying of ~200,000 files with 70GB in total, the process gets interrupted and the following message in dom0 is shown: "qfile-unpacker: Fatal error: Child process exited abnormally (error type: Success)". It's reproducable but I cannot figure out at which point the process dies. Sometimes I get 15 GB copied another time 45GB.

**To Reproduce**
1. Create 200,000 files with amount of 70 GB in total.
2. Copy them with qvm-copy-to-vm to another vm (with enough space available). I'm using a dvm as target and a AppVm as source
3. Files get partially copied and a message appears "qfile-unpacker: Fatal error: Child process exited abnormally (error type: Success)"

**Expected behavior**
1. No error message if copy was successfull.
2. Additionally copying should be reliable even with big amount of files

**Actual behavior**
Only some part of all files is copied and this message appears: "qfile-unpacker: Fatal error: Child process exited abnormally (error type: Success)"

**Additional context**
Before creating a dvm as source I'm adjusting the size of the named dvm by:
qvm-volume resize --verbose dvmWithbigVolume:private 90GB
afterwards I create a named dvm and start it freshly.

**Solutions you've tried**
Tried to run it several > 10x. Every time it breaks but at different points in time and in progress.

**Comment by iamahuman** [comment]
Is VM-to-VM copy also affected?

**Comment by 100111001** [comment]
Sorry, don't understand the question. 
I'm using qvm-copy-to-vm within a vm and copying to another vm. The source vm is a dvm and the target a AppVm.

Just to mention. For now I never made it once  to copy the whole 70GB. Every time the process was interrupted in between.

**Comment by iamahuman** [comment]
I misread your ticket. My apologizes.

**Comment by pwmarcz** [comment]
I tried but could not reproduce this so far.

- I tried AppVM to AppVM copy, and also copying to a disposable VM (with storage limit set to 90 GB)
- I created 200,000 files, 350KB each (`dd if=/dev/zero`)
- I copied them using `qvm-copy data/` (where `data` is the directory with all the files)

The `qfile-unpacker` program does not seem to leak memory or file descriptors.

I thought that maybe this is about the inode limit, but on reaching the inode limit the copy fails with a proper error message (No space left on device).
